templates/common/_base/units/machine-config-daemon-pull.service

```diff
@@ -9,14 +9,20 @@ contents: |
   ConditionPathExists=/etc/pivot/image-pullspec
   ConditionPathExists=/var/lib/kubelet/config.json
   ConditionPathExists=!/usr/local/bin/machine-config-daemon
+  # Pivot should run after MCD image is pulled
   Before=machine-config-daemon-host.service
+  # Start pulling after network is up
+  Before=network-online.target
 
   [Service]
```

Comment:
Don't you still need `Type=oneshot` since there are multiple `ExecStart` lines?

Reply:
Oneshot services don't support restarts in systemd v233